classes/Cart.php

```diff
@@ -545,7 +545,11 @@ public function getCartRules($filter = CartRule::FILTER_ACTION_ALL, $autoAdd = t
      */
     public function getDiscounts()
     {
-        return CartRule::getCustomerHighlightedDiscounts($this->id_lang, $this->id_customer, $this);
+        return CartRule::getCustomerHighlightedDiscounts(
+            $this->id_lang,
+            $this->id_customer,
+            $this instanceof Cart ? $this : new Cart($this->id)
```

Comment:
+1 That's weird

Reply:
Ok I get it, the `getCustomerHighlightedDiscounts` expect a `Cart` instance, where this class is not a `Cart` instance but a `ClassCore` instance, you should change the `getCustomerHighlightedDiscounts` definition to use `CartCore` instead

Reply:
Fixed.